pkg/master/controller.go

```diff
@@ -81,8 +91,11 @@ func (c *Controller) Start() {
 		glog.Errorf("Unable to perform initial Kubernetes RO service initialization: %v", err)
 	}
 
-	c.runner = util.NewRunner(c.RunKubernetesService, c.RunKubernetesROService, repair.RunUntil)
-	c.runner.Start()
+	c.runnerPortals = util.NewRunner(c.RunKubernetesService, c.RunKubernetesROService, repairPortals.RunUntil)
+	c.runnerPortals.Start()
+
+	c.runnerServiceNodePorts = util.NewRunner(c.RunKubernetesService, c.RunKubernetesROService, repairNodePorts.RunUntil)
```

Comment:
Why do you need two runners?  You can put all of them in the same runner (runner runs multiple loops)